### ISSUE TITLE
Support more metadata types

### ISSUE THREAD

**Issue by scolladon** [mcarvin8/sfdx-decomposer#1]
Hi !

I really like what this project propose.

I think it could support those metadata as well: 
- Global Value Set Translation
- Standard Value Set Translation
- Translations
- Marketing App Extension

Let me know if I can contribute in any manner 👍

**Comment by mcarvin8** [comment]
Thank you @scolladon for the proposal! 

I have added support for Global Value Set Translation, Standard Value Set Translation, and Marketing App Extension. Please confirm these updates on your end.

I will work on Translations next. If you have any example meta files for Translations I could reference and test decomposing, that would be a big help!

**Comment by mcarvin8** [comment]
Something to note while I was testing out Standard/Global Value Set Translations, any translation meta file which still has a comment in the file will not be processed correctly. The files still get decomposed, but the `<translation>` part will not contain the comment. All translation meta file will need to have proper translations before you can decompose them.

Going based on this comment from the documentation.

![image](https://github.com/mcarvin8/sfdx-decomposer/assets/90224411/66385742-3c89-455c-9106-64676e20eaa8)

**Comment by scolladon** [comment]
I think the `translation` scenario is similar to the `profile` one or the `workflow` / `object` metadata

Here is a sample of a Translation object :
```xml
<?xml version="1.0" encoding="UTF-8"?>
<Translations xmlns="http://soap.sforce.com/2006/04/metadata">
    <bots>
        <fullName>BotName</fullName>
        <botVersions>
            <botDialogs>Dialogs</botDialogs>
            <fullName>DialogName</fullName>
        </botVersions>
    </bots>
    <customApplications>
        <label>Application Label!</label>
        <name>Application</name>
    </customApplications>
    <customLabels>
        <label>Value</label>
        <name>Label</name>
    </customLabels>
    <customPageWebLinks>
        <label>WebPageLinkLabel!</label>
        <name>WebPageLink</name>
    </customPageWebLinks>
    <customTabs>
        <label>Tab Label!</label>
        <name>Tab</name>
    </customTabs>
    <flowDefinitions>
        <label>Flow Label!</label>
        <fullName>Flow</fullName>
        <flows>
            <fullName>Flow Version</fullName>
            <label>Flow version Label</label>
        </flows>
    </flowDefinitions>
    <pipelineInspMetricConfigs>
        <label>PipelineMetric Label!</label>
        <name>PipelineMetric</name>
    </pipelineInspMetricConfigs>
    <prompts>
        <label>Prompt Label!</label>
        <desctiption>Prompt Description</desctiption>
        <name>Prompt</name>
        <promptVersions>
            <label>Prompt Version Label</label>
            <name>Prompt Versions</name>
        </promptVersions>
    </prompts>
    <quickActions>
        <label>QuickAction Label!</label>
        <name>QuickAction</name>
        <aspect>Master</aspect>
    </quickActions>
    <reportTypes>
        <label>Report Type Label!</label>
        <desctiption>Report Type Description</desctiption>
        <name>Report Type</name>
        <sections>
            <columns>
                <label>Column Label</label>
                <name>Column Name</name>
            </columns>
            <label>Section Label</label>
            <name>Section Name</name>
        </sections>
    </reportTypes>
    <scontrols>
        <label>SControl Label!</label>
        <name>SControl</name>
    </scontrols>
</Translations>
```

**Comment by mcarvin8** [comment]
@scolladon - Thanks for the example.  I went in and added support for the Translations type in the `translations` folder using the example you provided. Let me know if this looks good to you.

My original confusion stems from the fact that I have not worked with `Translations` before and my current team's metadata folder didn't have a translations folder.

**Comment by mcarvin8** [comment]
Just a heads-up @scolladon . I'm in the process of converting this tool from Python to Typescript in order to add this tool as a 3rd party Salesforce CLI plugin.

Repo - https://github.com/mcarvin8/sfdx-decomposer-plugin

Currently have it testing SAT on the 18/20 metadata types supported by this Python script. I need to do additional work on the recursion factor needed to support `Bot` and `BotVersion` in the plugin project, but I might push support for those 2 types until I get an initial release pushed.

Need to just work on creating the test scripts and anything else necessary to get the initial build shipped (first time building a CLI plugin or writing Typescript so it's definitely been a learning curve).

**Comment by scolladon** [comment]
That is great !
I'd be very happy to contribute !
I could show you what we have done with sfdx-git-delta around test, CI/CD, pipeline and deployment

**Comment by mcarvin8** [comment]
Appreciate the offer @scolladon! Definitely wouldn't mind seeing how you test and deploy your plugin so I can see what to do with mine.

I can start some threads over on the actual plugin repo for some suggestions for contributing to my plugin. Some things I've noticed while testing so far which definitely should be addressed is:

1. Re-add support for Marketing App Extension. Something I missed with Marketing App Extension when adding it to this Python script was the Character Data (CDATA) that are in between certain tags such as Action Schema and Action Params (ref: https://developer.salesforce.com/docs/atlas.en-us.api_meta.meta/api_meta/meta_marketingappextension.htm?q=marketing+app#subtype_MarketingAppExtAction). I've been trying various methods to retain CDATA when the file gets decomposed and re-composed, but I've been having issues with the XML2JS parser. 

2. Look into how to preserve HTML entities when decomposing files. I had to add a function to compose command to re-convert symbols like & back to the HTML entity `&amp;`, but this is definitely not the most efficient manner. If we can add ensure the decompose function maintains the original HTML entities, we can remove these functions from the compose function.

3. Re-add support for Bot and Bot version. I had this working in this Python repo, so this shouldn't be too complex to re-add these types.

4. Updating the test classes. I just pushed initial test classes which are passing right now, but I'm open to suggestions on improving these tests or adding more. The tests currently are just running through all supported metadata types and confirming the expected output messages appear for each type.

5. And lastly, just getting this initial plugin release built and shipped. It's testing SAT on the remaining metadata types (17 of the original 20), so I think we can ship out this initial plugin release and then work on adding these items above.
